tools/build-toolchain.sh

```diff
@@ -92,25 +159,64 @@ make all-gcc -j "$JOBS"
 make all-target-libgcc -j "$JOBS"
 make install-gcc || sudo make install-gcc || su -c "make install-gcc"
 make install-target-libgcc || sudo make install-target-libgcc || su -c "make install-target-libgcc"
+echo "Finished Compiling GCC-$GCC_V for MIPS N64 - (pass 1) outside of the source tree"
 
-# Compile newlib
+echo "Compiling newlib-$NEWLIB_V"
 cd ../"newlib-$NEWLIB_V"
-CFLAGS_FOR_TARGET="-DHAVE_ASSERT_FUNC" ./configure \
+CFLAGS_FOR_TARGET="-DHAVE_ASSERT_FUNC -O2" ./configure \
   --target=mips64-elf \
   --prefix="$INSTALL_PATH" \
   --with-cpu=mips64vr4300 \
   --disable-threads \
   --disable-libssp \
   --disable-werror
 make -j "$JOBS"
-make install || sudo env PATH="$PATH" make install || su -c "env PATH=\"$PATH\" make install"
+make install || sudo env PATH="$PATH" make install || su -c "env PATH=\"$PATH\" make install" # Perhaps use `checkinstall` instead?!
+make distclean # Ensure we can build it again
+echo "Finished Compiling newlib-$NEWLIB_V"
+
+if [ "$BUILD" != "$HOST" ]; then
+  INSTALL_PATH="${FOREIGN_INSTALL_PATH}"
+fi
+
+if [ "$BUILD" != "$HOST" ]; then
+  echo "Compiling newlib-$NEWLIB_V for foreign host"
```

Comment:
Newlib is a target library. Why should we build it twice? It should be sufficient to build it once with either the first compiler

Reply:
It was not in the output (binary folder) expected when I built on WSL. I probably still have this wrong, but working on it...